### DISCUSSION TITLE
Achievement check to see what is missing in a collectable achievement

### DISCUSSION THREAD

**Discussion by Fridge-Box** [RetroAchievements/RAIntegration#919] [Ideas]
Hello! I'd like to ask for a way for players to view bitflags while playing, or allow *view only* for specific achievements.
For example, https://retroachievements.org/achievement/155198. This is a large list of collectables and there is no way for an ordinary player (except by viewing code notes and then *guessing*) to see what possible items aren't collected. It would be great for players to be able to inspect their memory and see what items are missing without it being considered cheating. It would be an amazing QoL update and provide great usability to retro games. Here's a small, very poor mock up to show the list of bitflags/items shown
https://imgur.com/CpP6vmy

**Comment by Jamiras** [comment]
I'm going to point out that we specifically disabled the ability to view logic at runtime for situations like this. You shouldn't be inspecting your memory, viewing code notes, or otherwise trying to decipher the logic.

RetroArch doesn't provide any way to view the runtime logic, and you have to have a specific set of knowledge to extract the data from the tools in the standalone emulators. This was felt to be an unfair advantage for those who were able to do it, so it was disabled for all users.

I'm embedding the provided image so it doesn't get lost if imgur expires it.
![image](https://user-images.githubusercontent.com/32680403/192402248-55b8557d-313b-4591-83a1-1c5ef6c9110f.png)

As long as it's implemented in some way where it's accessible to all players, it's a reasonable suggestion, but this sort of thing isn't going to scale well when the collectible list is hundreds of items long.

The Measured feature helps to track how close a player is to the goal (before it was implemented players only knew they were still missing something, but not how much they were missing). But it still doesn't help identify exactly which items a player has missed. In your suggestion, you're effectively creating smaller measured bars for subgoals within the achievement. It's still a guessing game for the user to figure out which of the hidden 1ups in level 1 they've missed, instead of which 4 of the 10 total they've missed.

Other platforms have player created guides to help with this. I see no reason why a user (or even the dev) couldn't just list the location of the items in the achievement comments. The player would still have to keep track of which ones they've found.

**Reply by Blazekickn** [reply]
personally i don't really consider wandering around the map aimlessly for hours looking for the one bitflag i forgot to flip very fun or engaging, i would love to have a feature like this for my sets

**Reply by Blazekickn** [reply]
obviously its up to the achievement developer whether or not they want to adopt detailed measured, and for segmented achievements like "Get all the 1-Ups in World 1" its fine to not have it, but for full game achievements like the pokemon one it'd be nice

**Reply by Blazekickn** [reply]
> this sort of thing isn't going to scale well when the collectible list is hundreds of items long.

Id want it so the achievement list stays how it is, but you select the achievement and then it shows you the detailed list. Having a giant list take up the entire achievements menu by default would not be ideal.

**Reply by Maximdraco** [reply]
I think it could even be shown after a certain percentage, and it don't need to be very detailed, in the case of pokemon it could be, alfter getting the 90% of items, say: hidden items (x/y), items (a/b).

The player will still have to find the items, but it could give hints. Or even that, a list of hints insteed of giving information directly

**Comment by Jamiras** [comment]
Somewhat related rant: There was a discussion recently about how many achievement sets don't have missables marked. At what point does the handholding overtake the sense of accomplishment and an achievement simply become a participation award?
